src/TestCentric/components/Elements/IControlElement.cs

```diff
@@ -29,6 +29,8 @@ namespace TestCentric.Gui.Elements
     /// <summary>
     /// IControlElement is implemented by elements that wrap controls.
     /// </summary>
+    // TODO: This should not exist since it exposes the fact that the
+    // underlying gui element is a control.
     public interface IControlElement : IViewElement
```

Comment:
It may not be that bad. In fact, I'm having trouble getting rid of IToolStripElement, which has the same problem. The issue for me is that ToolStripItems, Controls and MenuItems in windows are all in separate hierarchies. As an example, you can implement a menu with MenuItems or with ToolStripItems. I'd kind of like the two to use the same interface.

Note that I'm not talking about getting rid of ControlElement but only IControlElement. I'd kind of like to keep the details in the implementing classes, not the interfaces used by the presenters.

No guarantee I can do this, however.